src/java.base/share/classes/sun/net/www/http/HttpClient.java

```diff
@@ -901,6 +901,11 @@ private boolean parseHTTPHeader(MessageHeader responses, ProgressSource pi, Http
                         /* default should be larger in case of proxy */
                         keepAliveConnections = p.findInt("max", usingProxy?50:5);
```

Comment:
Hello Michael, should we do something similar for this `max` parameter value too and (re)set the `keepAliveConnections` to the defaults, if the server sends a negative value?

Reply:
Yes, that's a good suggestion. Will do.